### DISCUSSION TITLE
Inquiry on Path Length Calculation from tx to rx

### DISCUSSION THREAD

**Discussion by YuruZhang22** [NVlabs/sionna#507] [Q&A]
Hello,

I would like to inquire about the correct method to obtain the length of a path, i.e., the distance from the transmitter (tx) to the receiver (rx). Currently, we are using the function _paths._vertices_ to get the coordinates of each reflection point (vertex) in each path and then calculating the total path length. Is this the correct approach, or is there another recommended method?

However, we have encountered an issue. In our scenario, rays reflect off very distant buildings even when the TX and RX are close to each other. For example, in our scenario, the TX position is (-511, -367, 34), and the RX position is (-420, -292, 1). We calculated the first reflection point to be at (-71.42, -262.48, 15). The distance between TX and RX is only 122 meters, but the ray jumps to 453 meters away and then back. We do not understand why this happens. We have tried multiple sets of TX and RX positions and encountered the same problem.

Any insights or suggestions would be greatly appreciated.

Best,
Yuru Zhang

**Comment by faycalaa** [comment]
Hi @YuruZhang22 ,

Yes, using `Paths.vertices` is a valid approach to compute the total paths' length.

Regarding the issue you have encountered, could you please share a minimal code snippet that reproduces the problem?

**Reply by YuruZhang22** [reply]
Hi,

Blow is the function we use. This function is used to get the overall length of each path. _all_hops_ are the vertices location. The` last_loc = np.array(tx_position) ` here is (-511, -367, 34), and `rx_loc` is (-420, -292, 1).

```py
def get_distances(scene, paths):
    # get per-path distance, correspond to a
    all_hops = paths._vertices.numpy()
    dims = all_hops.shape # see paths line # 833: [max_depth, num_targets, num_sources, max_num_paths, 3], tf.float : Positions of intersection points.
    max_paths = dims[-2]
    max_rxs = dims[1]
    max_depth = dims[0]
    distances = np.zeros((max_rxs,max_paths))
    rx_name = sorted(list(scene.receivers)) # TODO check if consistent with paths
    for r in range(max_rxs):
        for p in range(max_paths):
            if np.sum(np.abs(all_hops[:,r,0,p,:])) == 0:  # if this path does not exist
                continue
            # tx position
            last_loc = np.array(tx_position) 

            rx_loc = scene.receivers.get(rx_name[r]).position.numpy() # rx loc
            diss = np.zeros(max_depth)
            for o in range(max_depth): # this is max_depth

                if np.sum(np.abs(all_hops[o,r,0,p,:])) == 0: 
                    continue  # if this hop does not exist in this path

                diss[o] = np.linalg.norm(last_loc - all_hops[o,r,0,p,:])
                last_loc = all_hops[o,r,0,p,:]

            distances[r,p] = sum(diss) + np.linalg.norm(last_loc - rx_loc) 

    return distances
```

**Reply by faycalaa** [reply]
Could you share a standalone code snippet that reproduces the issue? We need not only a function but a minimal, self-contained code that we can run "out-of-the-box" to reproduce the problem.
